src/Symfony/Contracts/Service/ServiceSubscriberTrait.php

```diff
@@ -67,10 +99,28 @@ public function setContainer(ContainerInterface $container): ?ContainerInterface
     {
         $this->container = $container;
 
+        if ($container instanceof ServiceProviderInterface) {
+            // TODO: what if this isn't an instance of ServiceProviderInterface?
+            foreach (\array_keys($container->getProvidedServices()) as $key) {
+                // change property from "uninitialized" to "unset"
+                unset($this->$key);
+            }
+        }
+
         if (method_exists(get_parent_class(self::class) ?: '', __FUNCTION__)) {
             return parent::setContainer($container);
         }
 
         return null;
     }
+
+    public function __get(string $name): mixed
+    {
+        // TODO: ensure cannot be called from outside of the scope of the object?
+        // TODO: what if class has a child/parent that allows this?
+        // TODO: call parent::__get()?
+        // TODO: how to handle nullable services?
```

Comment:
What's the issue with them? We should just fetch them *if they exist* and set them to null if not?